src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs

```diff
@@ -18,7 +18,11 @@ public static ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTask<
         public static ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implementation, CancellationToken cancellationToken)
             => BrokeredServiceBase.RunServiceImplAsync(implementation, cancellationToken);
 
+        [Obsolete("Use RunWithSolutionAsync instead", error: false)]
         public static ValueTask<Solution> GetSolutionAsync(this PythiaPinnedSolutionInfoWrapper solutionInfo, ServiceBrokerClient client, CancellationToken cancellationToken)
             => RemoteWorkspaceManager.Default.GetSolutionAsync(client, solutionInfo.UnderlyingObject, cancellationToken);
+
+        public static ValueTask<T> RunWithSolutionAsync<T>(this PythiaPinnedSolutionInfoWrapper solutionInfo, ServiceBrokerClient client, Func<Solution, ValueTask<T>> implementation, CancellationToken cancellationToken)
```

Comment:
Can these also be overloads of RunServiceAsync?

Reply:
sure

Reply:
done.